### ISSUE TITLE
Option to Disable Hover

### ISSUE THREAD

**Issue by drgrib** [sourcegraph/go-langserver#312]
The plug-in doesn't seem to respect global hover settings. I don't need hover and it is really invasive to my coding process, especially when it pops up for the 10th time I fill in `fmt.Println` to explain to me how the function works. And then continues to follow my cursor and take up a substantial amount of screen real estate as I try to focus on and remember what I was trying to type into the `Println`.

In short: I want autocomplete from the server, not hover. How do I accomplish this?

**Comment by keegancsmith** [comment]
the issue is with your client, not go-langserver. IE your client is sending hover requests, but it sounds like you don't want it to.

**Comment by drgrib** [comment]
@keegancsmith Okay. I'm using VS Code. But I have a feeling they would send me back to `go-langserver` for requests.

**Comment by slimsag** [comment]
Yeah, this is a problem with the client. The language server cannot do anything to reject such requests or access your hover settings, I think. I don't think they will defer you back here for this issue. @ramya-rao-a what do you think?

**Comment by ramya-rao-a** [comment]
@drgrib Add `"editor.parameterHints.enabled": false` to your settings.

For future reference, the presence and behavior of the UI element is the responsibility of the editor. The content in these elements is the responsibility of the editor extension. In the case of Go in VS Code, some of these are taken up by the language server

**Comment by drgrib** [comment]
@ramya-rao-a I'm afraid that does nothing for me. Where should I post about this problem? https://github.com/Microsoft/vscode-go?

**Comment by ramya-rao-a** [comment]
Yes, log an issue at https://github.com/Microsoft/vscode-go. Also include screenshots of what exactly you want to disable. We have different widgets that show up when typing: suggest widget, hover widget, parameter hint/signature help widget

**Comment by drgrib** [comment]
This actually solved it for me. Found it on StackOverflow:

```
"editor.parameterHints": false
```

Can't tell you how many frustrating, crippling moments of trying to edit Go in VS Code this caused me. Relieved to have it stopped.

**Comment by ramya-rao-a** [comment]
Interesting, in the Insiders build that setting is `editor.parameterHints.enabled` and that's why I suggested it. But in the stable build, the same is `editor.parameterHints`. I'll look into that separately.

Thanks for getting back as to what finally fixed it.
